ElectronClient/app/gui/NotebookPropertiesDialog.tsx

```diff
@@ -0,0 +1,66 @@
+import * as React from 'react';
+import { useState, useEffect } from 'react';
+
+const { _ } = require('lib/locale.js');
+const { themeStyle } = require('../theme.js');
+const DialogButtonRow = require('./DialogButtonRow.min');
+
+const Folder = require('lib/models/Folder.js');
+
+interface NotebookPropertiesDialogProps {
+	theme: number,
+	folderId: string,
+	onClose: Function,
+}
+
+export default function NotebookPropertiesDialog(props: NotebookPropertiesDialogProps) {
+	console.info('Render NotebookPropertiesDialog');
+
+	const [folder, formFolder] = useState<any>({});
+
+	const theme = themeStyle(props.theme);
+
+	useEffect(() => {
+		async function fetchFolder() {
+			if (!props.folderId) {
+				formFolder({folder: null});
+			} else {
+				formFolder(await Folder.load(props.folderId));
+			}
+		}
+
+		fetchFolder();
+	}, [props.folderId]);
+
+
+	const buttonRow_click = () => {
+		props.onClose();
+	};
+
+	const saveTitle = async () => {
+		await Folder.save({id: folder.id, title: folder.title, userValidation: true});
+	};
+
+	const saveIcon = async () => {
+		await Folder.save({id: folder.id, icon: folder.icon, userValidation: true});
+	};
+
+	const rootStyle = Object.assign({}, theme.dialogBox);
+	rootStyle.width = '50%';
+
+	return (
+		<div style={theme.dialogModalLayer}>
+			<div style={rootStyle}>
+				<div style={theme.dialogTitle}>{_('Notebook Properties')}</div>
+				<label style={theme.textStyle}>{_('Name:   ')}</label>
```

Comment:
Should use styling to align elements, especially since this is a string to be translated and the spacing would be different from one language to another.

Reply:
I will need to study a tiny bit on this. I do have work this afternoon and do need to run some errands beforehand in a bit.